src/utils.ts

```diff
@@ -1385,3 +1394,17 @@ export function emitWarningOnce(message: string): void {
     return emitWarning(message);
   }
 }
+
+/**
+ * Determine if a server supports retryable writes.
+ *
+ * @internal
+ */
+export function supportsRetryableWrites(server: Server): boolean | undefined | number {
```

Comment:
I think this can be tightened to boolean if we `!!` the `server.loadBalanced` and/or `server.description.logicalSessionTimeoutMinutes`, which is where I imagine the number and undefined could be coming from

Reply:
Ah bingo. This is updated now.